### PR TITLE
Feature: group command for easy read

### LINKED ISSUE #6324: How can I push chart to an insecure OCI registry with helm v3
How can I push chart to an insecure registry(HTTP or HTTPS with self-signed certificate) with helm v3?

Should we provide an option to skip the security checking? It's useful in developing/testing environments

### LINKED ISSUE #7903: should we group command for easy read?
if we can group command like `kubectl`,it will easy to read,like this
```
Release Commands:
  get         download extended information of a named release
  history     fetch release history
  install     install a chart
  list        list releases
  test        run tests for a release
  rollback    roll back a release to a previous revision
  status      displays the status of the named release
  template    locally render templates
  uninstall   uninstall a release
  upgrade     upgrade a release

Chart Commands:
  create      create a new chart with the given name
  dependency  manage a chart's dependencies
  pull        download a chart from a repository and (optionally) unpack it in local directory
  show        show information of a chart
  lint        examines a chart for possible issues
  package     package a chart directory into a chart archive
  repo        add, list, remove, update, and index chart repositories
  search      search for a keyword in charts
  verify      verify that a chart at the given path has been signed and is valid

Other Commands:
  completion  Generate autocompletions script for the specified shell (bash or zsh)
  env         Helm client environment information
  plugin      install, list, or uninstall Helm plugins
  version     print the client version information

Use "helm <command> --help" for more information about a given command.

```

### PR DESCRIPTION
**Pull request by zgfh**
group command for easy read 
Fix #7903

### CODE REVIEW

File: cmd/helm/root.go
Comment (waveywaves) [inline cmd/helm/root.go:138]: Can you change this from groups to commandGroups ?

File: cmd/helm/root.go
Comment (waveywaves) [inline cmd/helm/root.go:163]: This should be put in Release Commands.

File: cmd/helm/root.go
Comment (waveywaves) [inline cmd/helm/root.go:157]: Beginner and Intermediate Chart Commands sounds confusing regarding wrt user experience. I think you could try a little more on segregating this commands. Maybe based on repo usage, then chart verification and packaging maybe.

**Review by waveywaves — CHANGES_REQUESTED** [review]

File: cmd/helm/root.go
Comment (zgfh) [reply cmd/helm/root.go]: sorry, my bad.  just group release and charts is ok?

File: cmd/helm/root.go
Comment (bacongobbler) [inline cmd/helm/root.go:140]: I would re-word this as "Release Management Commands:". That way it's more descriptive about when you should use these commands.

File: cmd/helm/root.go
Comment (bacongobbler) [inline cmd/helm/root.go:151]: `helm template` might not be a good fit here, as it doesn't interact with the release ledger. Perhaps it should be placed under the other group.

File: cmd/helm/root.go
Comment (bacongobbler) [inline cmd/helm/root.go:159]: Some of these sub-commands don't deal with charts, but rather with chart repositories. Perhaps we could put these under a third group called "Chart Repository Commands". What do you think about that?

I'm thinking specifically

- `helm repo`
- `helm search`
- `helm pull`
- `helm show`

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by technosophos** [comment]
We have been told by the Kubernetes team that the `kuebctl` libraries are not to be used as dependencies. They are not "API stable" and will undergo breaking changes. We have been working to remove all of the `kubectl` code from Helm at their suggestion.

So my guess is that we cannot accept this PR simply on the grounds that upstream Kubernetes says not to do this.

File: cmd/helm/root.go
Comment (zgfh) [reply cmd/helm/root.go:151]: i agree too, @waveywaves what do you think? ,should me change back?

**Comment by zgfh** [comment]
@technosophos  good to know , let me close this, but it may be a time to talk about how group command

**Comment by technosophos** [comment]
@zgfh I talked to @adamreese and he says we can import `kubectl` now, but not `kubernetes/kubernetes`. So if you would like, we can re-open this.

**Comment by zgfh** [comment]
@technosophos  sure, and update,now it look like this
```
Release Management Commands:
  install     install a chart
  list        list releases
  get         download extended information of a named release
  status      displays the status of the named release
  upgrade     upgrade a release
  history     fetch release history
  rollback    roll back a release to a previous revision
  test        run tests for a release
  uninstall   uninstall a release

Chart Commands:
  create      create a new chart with the given name
  dependency  manage a chart's dependencies
  package     package a chart directory into a chart archive
  lint        examines a chart for possible issues
  verify      verify that a chart at the given path has been signed and is valid

Chart Repository Commands:
  repo        add, list, remove, update, and index chart repositories
  search      search for a keyword in charts
  pull        download a chart from a repository and (optionally) unpack it in local directory
  show        show information of a chart

Other Commands:
  completion  Generate autocompletions script for the specified shell (bash or zsh)
  env         Helm client environment information
  plugin      install, list, or uninstall Helm plugins
  template    locally render templates
  version     print the client version information

```

File: cmd/helm/root.go
Comment (zgfh) [reply cmd/helm/root.go]: 👍，updated

File: cmd/helm/root.go
Comment (zgfh) [reply cmd/helm/root.go]: update

File: cmd/helm/root.go
Comment (bacongobbler) [inline cmd/helm/root.go:182]: Can you please move this under "Chart Commands"?

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by technosophos** [comment]
@zgfh Thanks, btw, for your patience on this one. My fault for making this extra confusing.

File: cmd/helm/root.go
Comment (zgfh) [reply cmd/helm/root.go]: updated

**Comment by zgfh** [comment]
@bacongobbler have a look

**Review by bacongobbler — APPROVED** [review]
Code LGTM. I'll hold off on merging so other maintainers can have a look. If there's no further feedback by next week, ping me and I'll go ahead and merge. Thanks @zgfh.

**Comment by zgfh** [comment]
@bacongobbler ping

**Comment by liuming-dev** [comment]
How to show *Global Flags*, I cannot find it anywhere? The following help messages seem to be lost.

```
Global Flags:
      --add-dir-header                   If true, adds the file directory to the header
      --alsologtostderr                  log to standard error as well as files
      --debug                            enable verbose output
      --kube-apiserver string            the address and the port for the Kubernetes API server
      --kube-context string              name of the kubeconfig context to use
      --kube-token string                bearer token used for authentication
      --kubeconfig string                path to the kubeconfig file
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --log-file string                  If non-empty, use this log file
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
      --logtostderr                      log to standard error instead of files (default true)
  -n, --namespace string                 namespace scope for this request
      --registry-config string           path to the registry config file (default "/Users/apple/Library/Preferences/helm/registry.json")
      --repository-cache string          path to the file containing cached repository indexes (default "/Users/apple/Library/Caches/helm/repository")
      --repository-config string         path to the file containing repository names and URLs (default "/Users/apple/Library/Preferences/helm/repositories.yaml")
      --skip-headers                     If true, avoid header prefixes in the log messages
      --skip-log-headers                 If true, avoid headers when opening log files
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
  -v, --v Level                          number for the log level verbosity
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
```

**Comment by zgfh** [comment]
@liuming-dev  thanks for help,sorry for this, I find kubectl have a options to do this,let me fix
```
Usage:
  kubectl [flags] [options]

Use "kubectl <command> --help" for more information about a given command.
Use "kubectl options" for a list of global command-line options (applies to all commands).
```

**Comment by zgfh** [comment]
@liuming-dev have a look https://github.com/helm/helm/pull/8095

**Comment by marckhouzam** [comment]
It seems there are other side-effects to this change affecting the help text.  They are minor, as it is just the help text, but I feel we should at least be aware of them.  Fundamentally, the kubectl package used to group commands brings in different formatting for the help text.  This maybe configurable in the package, I don't know.

Not that we are now aligned with `kubectl`'s help output.

What I noticed in the help texts (some changes are nice):
1. PersistentFlags are not shown but relegated to different `helm <subcommands> options` commands, which haven't been added.  We use persistent flags for the root command (#8095) and for the `status` command (#8149).
1. The "Flags" sections is now renamed "Options"
1. The "Usage" section now shows an `[options]` suffix, which should be replacing the `[flags]` suffix.  However, this `[flags]` suffix needs to be turned off in helm. **This can be fixed using Cobra's `command.DisableFlagsInUseLine` field.**
1. The "Usage" section is moved lower.
1. All flags no longer show their type, but are now shown with their default value
1. The flag descriptions are no longer neatly aligned
1. The `-h, --help` flag is no longer shown
1. The `Usage` section is removed for commands with sub-commands (e.g., `helm repo -h`)
1. The completions acceptance-tests broke because they parse the help text (easy to fix)

**Comment by bacongobbler** [comment]
Thank you for the research @marckhouzam. I should’ve verified this PR manually before merging.

Given the amount of behavioural changes introduced by this PR, we should probably pull this out and restart from square one, rather than play catch-up over multiple pull requests.

If someone else doesn’t get there sooner, I’ll go ahead and contribute a PR Tuesday to revert this PR and close #8095 (Monday’s a Canadian federal holiday for Marc and I).

**Comment by zgfh** [comment]
@marckhouzam thanks a lot to fix this 
@bacongobbler sorry for this PR and broken the code,I should do more test like marckhouzam, please do anything to fix this

**Comment by zgfh** [comment]
@marckhouzam @bacongobbler  I will try to fix this:
1. change `status` to  cmd.flag as other command: https://github.com/helm/helm/commit/c2069f7e8b95d6817878ef1444864aafb1d3c6c4
3. add `DisableFlagsInUseLine: true` for all command 
9. acceptance-tests 

but I don't know this is we want ,or we should just change root command?

**Comment by marckhouzam** [comment]
> Thank you for the research @marckhouzam. I should’ve verified this PR manually before merging.

To be fair I had a look at the PR as well and didn't even consider it would affect other commands. The only reason I noticed is the acceptance-testing failure.

> Given the amount of behavioural changes introduced by this PR, we should probably pull this out and restart from square one, rather than play catch-up over multiple pull requests.

Maybe that is safer. Although I do like the command grouping. Hopefully we can get it back in soon after.

> If someone else doesn’t get there sooner, I’ll go ahead and contribute a PR Tuesday to revert this PR and close #8095 (Monday’s a Canadian federal holiday for Marc and I).

🇨🇦 🎉

**Comment by zgfh** [comment]
@marckhouzam @bacongobbler revert PR:  https://github.com/helm/helm/pull/8153
very sorry again

**Pull request merged**
